tests/api_resources/test_file.py

```diff
@@ -2,13 +2,23 @@
 
 import tempfile
 
+import pytest
+
 import stripe
 
 
 TEST_RESOURCE_ID = 'file_123'
 
 
 class TestFile(object):
+    @pytest.fixture
+    def setup_upload_api_base(self):
+        stripe.upload_api_base = stripe.api_base
+        stripe.api_base = None
+        yield
+        stripe.api_base = stripe.upload_api_base
+        stripe.upload_api_base = 'https://files.stripe.com'
```

Comment:
I'm not too familiar with the framework here, but are you sure this doesn't need to be in a `finally` to guarantee that these values are set back in case of an exception?

Reply:
Just tested it, and it's not necessary. The framework already catches exceptions in tests and reports them as errors, and still executed the post-`yield` code.